homeassistant/components/sensor/gearbest.py

```diff
@@ -0,0 +1,122 @@
+"""
+Parse prices of a item from gearbest.
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/sensor.gearbest/
+"""
+import logging
+import asyncio
+from datetime import timedelta
+
+import voluptuous as vol
+
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+import homeassistant.helpers.config_validation as cv
+from homeassistant.util import Throttle
+from homeassistant.helpers.entity import Entity
+
+REQUIREMENTS = ['gearbest_parser==1.0.4']
+_LOGGER = logging.getLogger(__name__)
+
+CONF_ITEMS = 'items'
+CONF_CURRENCY = 'currency'
+
+ICON = 'mdi:coin'
+MIN_TIME_BETWEEN_UPDATES = timedelta(seconds=2*60*60)  #2h * 60min * 60sec
+
+_ITEM_SCHEMA = vol.Schema({
+    vol.Optional("url"): cv.string,
+    vol.Optional("id"): cv.string,
+    vol.Optional("name"): cv.string,
+    vol.Optional("currency"): cv.string
+})
+
+_ITEMS_SCHEMA = vol.Schema([_ITEM_SCHEMA])
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_ITEMS): _ITEMS_SCHEMA,
+    vol.Required(CONF_CURRENCY): cv.string,
+})
+
+
+@asyncio.coroutine
+def async_setup_platform(hass, config, async_add_devices, discovery_info=None):
+    """Set up the Gearbest sensor."""
+
+    del discovery_info  #unused
+
+    hass.loop.run_in_executor(None, _add_items, hass, config,
```

Comment:
Don't do this either. If your setup is sync, don't define an async setup platform method.